modules/apps/transfer/keeper/export_test.go

```diff
@@ -26,3 +26,8 @@ func (k Keeper) GetAllDenomTraces(ctx sdk.Context) []types.DenomTrace {
 
 	return traces
 }
+
+// ConstructTokenFromCoin is a wrapper around constructTokenFromCoin for testing purposes.
+func (k Keeper) ConstructTokenFromCoin(ctx sdk.Context, coin sdk.Coin) (types.Token, error) {
```

Comment:
We have some different terminology to refer to the process of converting one thing into another: `PacketDataV1toV2` (converting packet data V1 to packet data V2), `ExtractDenomFromFullPath` (convert full path denom string to `Denom` object), `ConstructTokenFromCoin` (convert SDK `Coin` to `Token`). Maybe it would be nice to align on the wording: what about `CoinToToken`? Or `TokenFromCoin` if we also rename `PacketDataV1toV2` to `PacketDataV2FromV1`.

Actually now that I am looking at the implementation of `PacketDataV1toV2` I think we could move it to `packet.go` and refactor its signature to:

```go
func NewFungibleTokenPacketDataV2(packetData types.FungibleTokenPacketData) (types.FungibleTokenPacketDataV2, error)
```

or 

```go
func FromFungibleTokenPacketDataV1(packetData types.FungibleTokenPacketData) (types.FungibleTokenPacketDataV2, error)
```

Although this would make the function public. So maybe if we want to keep it internal, we could do the renaming but leave the function where it is right now.

Reply:
I'm personally fine with the different namings. `constructTokenFromCoin` is a keeper function so I find it falls into a different category since it isn't doing a pure conversion or unmarshal

Reply:
I made the change https://github.com/cosmos/ibc-go/pull/6426/commits/0f5c43fb8559be6b76e3c7894e3ad8bb479b222c